examples/github.js

```diff
@@ -0,0 +1,105 @@
+'use strict'
+
+const fastify = require('fastify')({ logger: { level: 'trace' } })
+const sget = require('simple-get')
+
+// const oauthPlugin = require('fastify-oauth2')
+const oauthPlugin = require('..')
+
+fastify.register(oauthPlugin, {
+  name: 'githubOAuth2',
+  scope: [],
+  credentials: {
+    client: {
+      id: '<CLIENT_ID>',
+      secret: '<CLIENT_SECRET>'
+    },
+    auth: oauthPlugin.GITHUB_CONFIGURATION
+  },
+  startRedirectPath: '/login/github',
+  callbackUri: 'http://localhost:3000/login/github/callback'
+})
+
+let refreshToken = ''
+
+fastify.get('/login/github/callback', function (request, reply) {
+  this.githubOAuth2.getAccessTokenFromAuthorizationCodeFlow(
+    request,
+    (err, result) => {
+      if (err) {
+        reply.send(err)
+        return
+      }
+
+      refreshToken = result.token.refresh_token
+
+      sget.concat(
+        {
+          url: 'https://api.github.com/user',
+          method: 'GET',
+          headers: {
+            Authorization: 'token ' + result.token.access_token
+          },
+          json: true
+        },
+        function (err, res, data) {
+          if (err) {
+            reply.send(err)
+            return
+          }
+          reply.send(data)
+        }
+      )
+    }
+  )
+})
+
+fastify.get('/login/github/verifyAccessToken', function (request, reply) {
+  const { accessToken } = request.query
+  sget.concat(
+    {
+      url: 'https://api.github.com/applications/<CLIENT_ID>/token',
+      method: 'POST',
+      headers: {
+        Authorization:
+          'Basic ' +
+          Buffer.from('<CLIENT_ID>' + ':' + '<CLIENT_SECRET').toString(
+            'base64'
+          )
+      },
+      body: JSON.stringify({ access_token: accessToken }),
+      json: true
+    },
+    function (err, res, data) {
+      if (err) {
+        reply.send(err)
+        return
+      }
+      reply.send(data)
+    }
+  )
+})
+
+fastify.get('/login/github/refreshToken', function (request, reply) {
+  // this.githubOAuth2.getAccessTokenFromRefreshToken(refresh_token, {}, (err, result) => {})
+
+  sget.concat(
```

Comment:
I would not use `sget` here.
See my comment in https://github.com/fastify/fastify-oauth2/issues/174#issuecomment-1220414854

Reply:
Thanks @climba03003